### ISSUE TITLE
Is BTRFS source and/or destination a problem?

### ISSUE THREAD

**Issue by ghost** [laurent22/rsync-time-backup#274]
Hi,

Is this script supposed to work with BTRFS formatted local disks as the backup source and the destination? Backing up about 750GB and it filled up a 4TB disk with only 5 backups. It looks like the hard-linking logic isn't working and each backup is a full backup. So, I'm wondering if it's because it's on BTRFS? I didn't have this problem before backing up other computers to vanilla EXT4 volumes. But I'm not sure whether the problem is because the source is BTRFS or the backup is BTRFS, or both? Or something entirely different... Thanks!

**Comment by shmilee** [comment]
Yes, it works with BTRFS. I've used a modified script based on rsync-time-backup for several months, with the same or different BTRFS partitions.
see: https://github.com/shmilee/arch-time-backup/tree/master?tab=readme-ov-file#an-actual-use-case

Maybe, your problem is related to the BTRFS mount options? rsync options?

**Comment by ghost** [comment]
> Yes, it works with BTRFS. I've used a modified script based on rsync-time-backup for several months, with the same or different BTRFS partitions. see: https://github.com/shmilee/arch-time-backup/tree/master?tab=readme-ov-file#an-actual-use-case
> 
> Maybe, your problem is related to the BTRFS mount options? rsync options?

Thanks! But is the source BTRFS too? The script backs up everything every time if the source is BTRFS. The destination doesn't matter, it seems. I'm gonna try to verify that today, I wiped and reinstalled on EXT filesystem, but didn't get to the script yet.

I used the vanilla tm backup script, I didn't change any rsync option or BTRFS options either as I don't know how to, just whatever the Disks utility did. The script worked for me in the past, but I never used anything besides EXT before.

In the end this is not a big deal, I can do EXT for boot, I'm just playing around with Linux at this time, it's just testing.

**Comment by ghost** [comment]
So yeah, on a different system now (Fedora 41 KDE) and source /home is on BTRFS volume and the script backs up everything each time it runs :(

**Comment by ghost** [comment]
I found another, similar script to yours, but simpler, and it does not have this issue. It backs up as expected, only changed and new files. Both source and destination are BTRFS volumes. I've tested both scripts yesterday for several hours. 

If you're curious, this is the one: https://github.com/cytopia/linux-timemachine

**Comment by psmanek** [comment]
TimeMachine script does not have retention feature. The only similarity is rsync and Apple TomeMachine.

**Comment by ghost** [comment]
> TimeMachine script does not have retention feature.

Yes, I know, it's simpler. I'd rather use this one, but it is not working for me as it backs up everything every time it runs when the source is on BTRFS volume. But I was wondering if it can be fixed. 

I used Mint on ext4 before but I'm on Fedora 41 now and BTRFS is the default FS.

**Comment by psmanek** [comment]
Both scripts use rsync. Check what parameters are used in second script and use them in first script as additional parameters.

**Comment by ghost** [comment]
Yes, I looked inside both scripts but this is way beyond my abilities, I can't figure this out.
